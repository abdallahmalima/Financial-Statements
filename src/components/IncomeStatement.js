import PropTypes from 'prop-types';
import currencyFormatter from '../utils/currencyFormatter';

const IncomeStatement = ({
  incomeStatement: { calendarYear, operatingExpenses, operatingIncome },
}) => (
<div className="w-full px-2 mb-4">
      <div className="bg-red-500 rounded-lg shadow-lg overflow-hidden">
        <div className="flex  p-3">
          <h3 className="text-2xl font-bold text-gray-900 mb-2">{calendarYear}</h3>
          <div className="flex flex-col p-3 ">
          <h4 className="text-l font-bold text-gray-900 mb-2">Operating Expenses:<span className="px-2 py-1 font-semibold rounded-full bg-red-600 text-white">{currencyFormatter(operatingExpenses)}</span></h4>
          <h4 className="text-l font-bold text-gray-900 mb-2">Operating Income:<span className="px-2 py-1 font-semibold rounded-full bg-red-600 text-white">{currencyFormatter(operatingIncome)}</span></h4>
          </div>

        </div>
      </div>
    </div>
);

IncomeStatement.propTypes = {
  incomeStatement: PropTypes.shape({
    calendarYear: PropTypes.string.isRequired,
    operatingExpenses: PropTypes.number.isRequired,
    operatingIncome: PropTypes.number.isRequired,
  }).isRequired,
};

export default IncomeStatement;