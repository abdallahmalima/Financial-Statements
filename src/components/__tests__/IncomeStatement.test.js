import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import IncomeStatement from '../IncomeStatement';

describe('Test Income Statement Component', () => {
  it('check if component component match the snapshot', () => {
    
   const incomeStatement={
        calendarYear: '2022',
        operatingExpenses: 10000,
        operatingIncome: 20000,
      };

    const incomeStatementRender = render(
        <Router>
          <IncomeStatement incomeStatement={incomeStatement} />
        </Router>
    );

    expect(incomeStatementRender).toMatchSnapshot();
  });


  it('check if component component match the snapshot', () => {
    
    const incomeStatement={
         calendarYear: '2022',
         operatingExpenses: 10000,
         operatingIncome: 20000,
       };
 
     const { getByText } = render(
         <Router>
           <IncomeStatement incomeStatement={incomeStatement} />
         </Router>
     );
 
     expect(getByText('2022')).toBeInTheDocument();
     expect(getByText('Operating Expenses:')).toBeInTheDocument();
     expect(getByText('Operating Income:')).toBeInTheDocument();
   });


});
