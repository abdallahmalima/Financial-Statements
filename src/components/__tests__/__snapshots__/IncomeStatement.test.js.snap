// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Financial Statement Component check if component component match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-1/2 px-2 mb-4"
      >
        <div
          class="bg-red-500 rounded-lg shadow-lg overflow-hidden"
        >
          <div
            class="flex flex-col p-5 items-center"
          >
            <h3
              class="text-2xl font-bold text-gray-900 mb-2"
            >
              ABC
            </h3>
            <button
              class="px-2 py-2 font-semibold rounded shadow-lg bg-red-700 text-gray-300 text-xl"
            >
              View More
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-1/2 px-2 mb-4"
    >
      <div
        class="bg-red-500 rounded-lg shadow-lg overflow-hidden"
      >
        <div
          class="flex flex-col p-5 items-center"
        >
          <h3
            class="text-2xl font-bold text-gray-900 mb-2"
          >
            ABC
          </h3>
          <button
            class="px-2 py-2 font-semibold rounded shadow-lg bg-red-700 text-gray-300 text-xl"
          >
            View More
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Income Statement Component check if component component match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full px-2 mb-4"
      >
        <div
          class="bg-red-500 rounded-lg shadow-lg overflow-hidden"
        >
          <div
            class="flex  p-3"
          >
            <h3
              class="text-2xl font-bold text-gray-900 mb-2"
            >
              2022
            </h3>
            <div
              class="flex flex-col p-3 "
            >
              <h4
                class="text-l font-bold text-gray-900 mb-2"
              >
                Operating Expenses:
                <span
                  class="px-2 py-1 font-semibold rounded-full bg-red-600 text-white"
                >
                  $10,000.00
                </span>
              </h4>
              <h4
                class="text-l font-bold text-gray-900 mb-2"
              >
                Operating Income:
                <span
                  class="px-2 py-1 font-semibold rounded-full bg-red-600 text-white"
                >
                  $20,000.00
                </span>
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full px-2 mb-4"
    >
      <div
        class="bg-red-500 rounded-lg shadow-lg overflow-hidden"
      >
        <div
          class="flex  p-3"
        >
          <h3
            class="text-2xl font-bold text-gray-900 mb-2"
          >
            2022
          </h3>
          <div
            class="flex flex-col p-3 "
          >
            <h4
              class="text-l font-bold text-gray-900 mb-2"
            >
              Operating Expenses:
              <span
                class="px-2 py-1 font-semibold rounded-full bg-red-600 text-white"
              >
                $10,000.00
              </span>
            </h4>
            <h4
              class="text-l font-bold text-gray-900 mb-2"
            >
              Operating Income:
              <span
                class="px-2 py-1 font-semibold rounded-full bg-red-600 text-white"
              >
                $20,000.00
              </span>
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Income Statement Component check if component component match the snapshot 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full px-2 mb-4"
      >
        <div
          class="bg-red-500 rounded-lg shadow-lg overflow-hidden"
        >
          <div
            class="flex  p-3"
          >
            <h3
              class="text-2xl font-bold text-gray-900 mb-2"
            >
              2022
            </h3>
            <div
              class="flex flex-col p-3 "
            >
              <h4
                class="text-l font-bold text-gray-900 mb-2"
              >
                Operating Expenses:
                <span
                  class="px-2 py-1 font-semibold rounded-full bg-red-600 text-white"
                >
                  $10,000.00
                </span>
              </h4>
              <h4
                class="text-l font-bold text-gray-900 mb-2"
              >
                Operating Income:
                <span
                  class="px-2 py-1 font-semibold rounded-full bg-red-600 text-white"
                >
                  $20,000.00
                </span>
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full px-2 mb-4"
    >
      <div
        class="bg-red-500 rounded-lg shadow-lg overflow-hidden"
      >
        <div
          class="flex  p-3"
        >
          <h3
            class="text-2xl font-bold text-gray-900 mb-2"
          >
            2022
          </h3>
          <div
            class="flex flex-col p-3 "
          >
            <h4
              class="text-l font-bold text-gray-900 mb-2"
            >
              Operating Expenses:
              <span
                class="px-2 py-1 font-semibold rounded-full bg-red-600 text-white"
              >
                $10,000.00
              </span>
            </h4>
            <h4
              class="text-l font-bold text-gray-900 mb-2"
            >
              Operating Income:
              <span
                class="px-2 py-1 font-semibold rounded-full bg-red-600 text-white"
              >
                $20,000.00
              </span>
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
